docs/adrs/2023-05-08-exposing-metrics.md

```diff
@@ -0,0 +1,93 @@
+# Exposing metrics
+
+Date: 2023-05-08
+
+**Status**: Proposed
+
+## Context
+
+Prometheus metrics are a common way to monitor the cluster. Providing metrics
+can be a helpful way to monitor scale sets and health of the ephemeral runners.
+
+## Proposal
+
+There are two main components driving the behaviour of the scale set:
+
+1. ARC controllers responsible for managing kubernetes resources.
+2. The `AutoscalingListener`, driver of the autoscaling solution responsible for
+   describing the desired state.
+
+To expose metrics, we would need to create 3 additional resources:
+
+1. `ServiceMonitor` - resource used by prometheus to match namespaces and
+    services from where it needs to gather metrics
+2. `Service` for the `gha-runner-scale-set-controller` - service that will
+    target ARC controller `Deployment`
+3. `Service` for the `gha-runner-scale-set` listeners - service that will target
+    all listeners for each `AutoscalingRunnerSet`
+
+Metrics can be enabled or disabled on the controller level. If the flag
+`metrics-addr` is not empty, the controller and listeners are going to register
+the appropriate collectors.
+
+Alternatively, if there is a need to control the metrics exposure per
+`AutoscalingRunnerSet`, we can control it by applying additional label on the
+listener.
+
+### Metrics exposed by the controller
+
+To get a better understanding about health and workings of the cluster
```

Comment:
i assume each metrics will be tagged with the RunnerScleSet id/name/url etc?

Reply:
I used labels that you assigned in POC which are great! We can always add or delete ones. That is the reason I said I'm not sure if we actually need to include them in the ADR. Changing labels or metrics would require an update on the ADR. But I wanted to display the basic structure and then trim it down if we need or extend it if you think we should